src/node.cc

```diff
@@ -2659,6 +2659,17 @@ static void EnvGetter(Local<Name> property,
 static void EnvSetter(Local<Name> property,
                       Local<Value> value,
                       const PropertyCallbackInfo<Value>& info) {
+  Environment* env = Environment::GetCurrent(info);
+  if (config_pending_deprecation && env->EmitProcessEnvWarning() &&
+      !value->IsString() && !value->IsNumber() && !value->IsBoolean()) {
+    if (ProcessEmitDeprecationWarning(
+          env,
+          "Assigning any value other than a string, number, or boolean value "
```

Comment:
Nit: I am not a native English speaker, but I believe this message can be presented better. Unfortunately, I am also not sure how to improve it.

Reply:
I'll remove the second "value" on landing. Hope that helps.

Reply:
Thanks. That felt redundant to me, when I read that for the first time.

Also, does it sound better if we change the second part to, "If the value is not one of them, use the string representation of it."?